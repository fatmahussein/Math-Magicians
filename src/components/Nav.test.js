import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from './Nav';

test('Navbar snapshot', () => {
  const { asFragment } = render(
    <Router>
      {' '}
      {/* Wrap component in BrowserRouter */}
      <Navbar />
    </Router>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test('Navbar link destinations', () => {
  const { getByText } = render(
    <Router>
      <Navbar />
    </Router>,
  );

  const homeLink = getByText('Home');
  const calculatorLink = getByText('Calculator');
  const quotesLink = getByText('Quotes');

  expect(homeLink.getAttribute('href')).toBe('/');
  expect(calculatorLink.getAttribute('href')).toBe('/Calculator');
  expect(quotesLink.getAttribute('href')).toBe('/Quotes');
});
