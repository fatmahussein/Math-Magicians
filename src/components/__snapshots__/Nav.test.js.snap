// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <nav>
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Home
      </a>
      <div
        class="border"
      />
      <a
        class=""
        href="/Calculator"
      >
         Calculator
      </a>
      <div
        class="border"
      />
      <a
        class=""
        href="/Quotes"
      >
         Quotes
      </a>
    </nav>
  </div>
</DocumentFragment>
`;
